### PR TITLE
ICMPv4 support for the socket stack

### PR DESCRIPTION
**Pull request by yomimono**
Provide preliminary ICMP support for the socket stack via unprivileged sockets. See https://lwn.net/Articles/420800/ or http://www.manpagez.com/man/4/icmp/ for more information.

### CODE REVIEW

**Comment by yomimono** [comment]
The socket ICMP test fails with `EACCES` on Travis.  It may be possible to make it pass by asking for the right Linux distribution.  FWIW tests do pass locally on OSX.

File: unix/icmpv4_socket.ml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+open Lwt.Infix
```
Comment (talex5) [inline unix/icmpv4_socket.ml]: `.mli` file missing from commit?

Comment (yomimono) [inline unix/icmpv4_socket.ml]: Indeed, and including it reveals a couple of needed fixes; thanks for pointing this out.

File: unix/icmpv4_socket.mli [hunk 2]
```diff
@@ -0,0 +1,5 @@
+include V1_LWT.ICMP with type ipaddr = Ipaddr.V4.t
+                     and type buffer = Cstruct.t
+                     and type 'a io = 'a Lwt.t
+
+val connect : unit -> unit io
```
Comment (talex5) [inline unix/icmpv4_socket.mli]: `unit -> t io`?

Comment (yomimono) [inline unix/icmpv4_socket.mli]: This PR needs more coffee.

**Comment by yomimono** [comment]
Since it seems there are no objections, I'll merge this once it's passing.  ISTM that even on platforms that don't support unprivileged sockets, this is an improvement over the complete lack of support that previously existed.

**Pull request merged**